srcjs/shinyapp.js

```diff
@@ -456,8 +456,13 @@ var ShinyApp = function() {
 
   // Adds custom message handler - this one is exposed to the user
   function addCustomMessageHandler(type, handler) {
+    // Remove any previously defined handlers so that only the most recent one
+    // will be called
     if (customMessageHandlers[type]) {
-      throw('handler for message of type "' + type + '" already added.');
+      var typeIdx = customMessageHandlerOrder.indexOf(type);
+      if (typeIdx != -1) {
```

Comment:
Can you use `!==` here? I just realized that our eslint rules didn't check for this, and I've enabled that check on master.